src/fhirtypes/ElementDefinition.ts

```diff
@@ -386,13 +386,27 @@ export class ElementDefinition {
     validationErrors.push(
       this.validateIncludes(slicing.rules, ALLOWED_SLICING_RULES, 'slicing.rules')
     );
-
+    const isR4 = this.structDef.fhirVersion.startsWith('4.');
     slicing.discriminator?.forEach((d, i) => {
       const discriminatorPath = `slicing.discriminator[${i}]`;
       validationErrors.push(this.validateRequired(d.type, `${discriminatorPath}.type`));
-      validationErrors.push(
-        this.validateIncludes(d.type, ALLOWED_DISCRIMINATOR_TYPES, `${discriminatorPath}.type`)
-      );
+      if (isR4) {
+        validationErrors.push(
+          this.validateIncludes(d.type, ALLOWED_DISCRIMINATOR_TYPES, `${discriminatorPath}.type`)
+        );
+      } else {
+        validationErrors.push(
+          this.validateIncludes(d.type, ALLOWED_DISCRIMINATOR_TYPES_R5, `${discriminatorPath}.type`)
+        );
+        if (d.type === 'position' && slicing.ordered !== true) {
+          validationErrors.push(
+            new ValidationError(
+              'Slicing ordering must be true when a position discriminator is used.',
+              'slicing.ordered'
+            )
+          );
+        }
```

Comment:
Here's something to think about: If the user uses a `position` discriminator type, should we just set `ordered` to true for them? I.e., we know what it should be, so... just do it? Putting the "short" in "shorthand"?

I'm torn on it. I feel like if they don't set the `ordered` property themself using a caret rule, then it makes sense to set it for them. But if they had set it themself to the _wrong_ value (false), then I'm not sure if silently setting it to true is the right thing to do...